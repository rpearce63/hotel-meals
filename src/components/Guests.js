import React, { Component } from "react";
import PropTypes from "prop-types";

export default class Guests extends Component {
  render() {
    const { names, dates } = this.props;
    return (
      <div>
        <textarea
          name="names"
          cols="30"
          rows="2"
          placeholder="Guest name"
          value={names}
          onChange={this.props.getGuestInfo}
          tabIndex="1"
        />
        <textarea
          name="dates"
          cols="30"
          rows="2"
          placeholder="Dates"
          value={dates}
          onChange={this.props.getGuestInfo}
          tabIndex="2"
        />
      </div>
    );
  }
}

Guests.propTypes = {
  names: PropTypes.string,
  dates: PropTypes.string,
  getGuestInfo: PropTypes.func
};
