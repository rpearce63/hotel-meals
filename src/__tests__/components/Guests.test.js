import React from "react";
import { shallow } from "enzyme";

import Guests from "../../components/Guests";

it("renders without crashing", () => {
  const wrapper = shallow(<Guests />);
  expect(wrapper).toMatchSnapshot();
});
