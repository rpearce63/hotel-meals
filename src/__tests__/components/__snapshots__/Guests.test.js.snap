// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Guests />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <textarea
          cols="30"
          id="namesText"
          name="names"
          placeholder="Guest name"
          rows="2"
          tabIndex="1"
        />,
        <textarea
          cols="30"
          id="datesText"
          name="dates"
          placeholder="Dates"
          rows="2"
          tabIndex="2"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "cols": "30",
          "id": "namesText",
          "name": "names",
          "onChange": undefined,
          "placeholder": "Guest name",
          "rows": "2",
          "tabIndex": "1",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "cols": "30",
          "id": "datesText",
          "name": "dates",
          "onChange": undefined,
          "placeholder": "Dates",
          "rows": "2",
          "tabIndex": "2",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <textarea
            cols="30"
            id="namesText"
            name="names"
            placeholder="Guest name"
            rows="2"
            tabIndex="1"
          />,
          <textarea
            cols="30"
            id="datesText"
            name="dates"
            placeholder="Dates"
            rows="2"
            tabIndex="2"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "cols": "30",
            "id": "namesText",
            "name": "names",
            "onChange": undefined,
            "placeholder": "Guest name",
            "rows": "2",
            "tabIndex": "1",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "cols": "30",
            "id": "datesText",
            "name": "dates",
            "onChange": undefined,
            "placeholder": "Dates",
            "rows": "2",
            "tabIndex": "2",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
